Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -362,6 +362,17 @@ ShaderCode GenerateVertexShaderCode(APIType api_type, const vertex_shader_uid_da
       break;
     }
 
+    // When q is 0, the GameCube appears to have a special case
+    // This can be seen in devkitPro's neheGX Lesson08 example for Wii
+    // Makes differences in Rogue Squadron 3 (Hoth sky) and The Last Story (shadow culling)
+    // TODO: check if this only affects XF_TEXGEN_REGULAR
```

Comment:
this comment seems bizarre especially with the condition right below it. shouldn't we check in both places, or not check at all?